src/streamlink/plugins/vimeo.py

```diff
@@ -25,6 +25,22 @@
     r"https?://(player\.vimeo\.com/video/\d+|(www\.)?vimeo\.com/.+)",
 ))
 class Vimeo(Plugin):
+    VIEWER_URL = "https://vimeo.com/_next/viewer"
+    OEMBED_URL = "https://vimeo.com/api/oembed.json"
+    API_URL = "https://{0}/{1}?fields=config_url"
```

Comment:
It's probably better to use the `streamlink.utils.url.update_scheme("https://", host)` method, adding the path via `urllib.parse.urljoin()` and setting the `fields` query param via the `params` parameter on the GET request directly.